src/Components/WebAssembly/JSInterop/src/WebAssemblyJSRuntime.cs

```diff
@@ -102,13 +101,26 @@ protected override void SendByteArray(int id, byte[] data)
                         ? throw new JSException(exception)
                         : (TResult)(object)new WebAssemblyJSObjectReference(this, id);
                 case JSCallResultType.JSStreamReference:
-                    // WebAssembly streaming interop is not presently supported (TODO). Thus we shouldn't see
-                    // a JSCallResultType.JSStreamReference.
+                    var serializedStreamReference = InternalCalls.InvokeJS<T0, T1, T2, string>(out exception, ref callInfo, arg0, arg1, arg2);
+                    return exception != null
+                        ? throw new JSException(exception)
+                        : (TResult)(object)DeserializeJSStreamReference(serializedStreamReference);
                 default:
                     throw new InvalidOperationException($"Invalid result type '{resultType}'.");
             }
         }
 
+        private IJSStreamReference DeserializeJSStreamReference(string serializedStreamReference)
+        {
+            var jsStreamReference = JsonSerializer.Deserialize<IJSStreamReference>(serializedStreamReference, JsonSerializerOptions);
```

Comment:
Do we have a custom reader for this? In that case, you could use the UnconditionalSuppressMessageAttribute and suppress the warning saying this is not doing reflection.

Reply:
Yes. I already put up:

https://github.com/dotnet/aspnetcore/pull/33986/files#diff-05d755169cd9aafe45f11033920fb3529925c0881ff6604dda66b48998f041a5R1-R11

Would that be sufficient, or did you have something else in mind?

Reply:
So the way it works is that XML file is only meant to ignore these warnings when we run the WasmLinkTest and is meant for us to be a reminder to fix it. The attribute goes further and says this is not a real issue

Reply:
Gotcha, appreciate the details! 😄

Reply:
We have https://github.com/dotnet/aspnetcore/blob/c01f9e3844f17925bb7f1ae99eac772d19692be0/src/JSInterop/Microsoft.JSInterop/src/Infrastructure/JSStreamReferenceJsonConverter.cs but it's in a different assembly, I assume that's not an issue for the unconditional suppression?